dask/array/slicing.py

```diff
@@ -1673,8 +1673,8 @@ def block_index_shape_from_1d_bool_index(dim, loc0, loc1):
         return np.sum(index[loc0:loc1])
 
     @functools.lru_cache
-    def n_preceeding_from_1d_bool_index(dim, loc0):
-        """Number of True index elements preceeding position loc0.
+    def n_preceding_from_1d_bool_index(dim, loc0):
```

Comment:
Is it OK to change the function name? It's not part of the API as far as I can see — `setitem_array` is missing from the documentation.

Reply:
I'd rather not change this, Dask and public API is a bit weird sometimes, even though I agree with you that nobody should be using it

Reply:
Perhaps duplicate the function (it's very short) and add a deprecation notice to the existing one?

Reply:
yep, you can just call the new function from the old one

Reply:
I call the old deprecated function from the new one - I suspect that's what you meant.